k4FWCore/scripts/k4run

```diff
@@ -45,62 +45,65 @@ FILTER_GAUDI_PROPS = [
     "AutoRetrieveTools",
 ]
 
-# ---------------------------------------------------------------------
 
-seen_files = set()
-option_db = {}
+def add_arguments(parser, app_mgr):
+    """
+        Add arguments to the parser for all properties of all configurables in the application manager
 
+        :param parser: the parser to add the arguments to
+        :param app_mgr: the application manager to get the properties from
+
+        :return: a dictionary mapping the argument name to the configurable it belongs to
+
+        Iterate over all the properties of all configurables in the application manager and add them to the parser.
+        The property name is used as the argument name (twice) and the property value as the default value.
+        If the property is a list, the type of the first element is used as the type of the argument.
+
+    """
+
+    option_db = {}
 
-def add_arguments(parser, app_mgr):
     # length increases when properties of an algorithm with tools are inspected
     # see https://github.com/key4hep/k4FWCore/pull/138
     # can contain the same value multiple times
     # see https://github.com/key4hep/k4FWCore/issues/141
     for conf in frozenset(app_mgr.allConfigurables.values()):
-        # skip public tools and the applicationmgr itself
-        if "ToolSvc" in conf.name() or "ApplicationMgr" in conf.name():
+        if conf.name() in ["ApplicationMgr", "ToolSvc", "k4FWCore__Sequencer", "k4FWCore__Algs"]:
             continue
-        # dict propertyname: (propertyvalue, propertydescription)
-        props = conf.getPropertiesWithDescription()
+        props = conf.getPropertiesWithDescription()  # property values -> property description
         for prop in props:
             if (
                 prop in FILTER_GAUDI_PROPS
                 or "Audit" in prop
                 or hasattr(props[prop][0], "__slots__")
             ):
                 continue
```

Comment:
Assuming `props` is a dictionary, this whole loop could be simplified via
```suggestion
        for prop_name, prop_values in props.items():
            if (
                prop_name in FILTER_GAUDI_PROPS
                or "Audit" in prop_name
                or hasattr(prop_values[0], "__slots__")
            ):
                continue
```
and then similar for the rest of the loop. (github's UI doesn't allow to easily make suggestions over areas with mixed diffs).

Reply:
Done